### PR TITLE
Update health check

### PR DESCRIPTION
**Pull request by nshumoogum**
### What

Updated the health check to use the latest version of `dp-healthcheck` and `dp-api-clients-go` libraries. This included updating the Makefile to properly extract the service version information.

Also updated graceful shutdown to follow a logical process if signal is sent to kill application.
 
### How to review

* Check changes make logical sense
* Unit tests pass
* No errors when signalling application to shutdown

### Who can review

Anyone

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -33,54 +39,69 @@ func main() {
...
+	if ctx.Err() == context.DeadlineExceeded {
+		log.Event(ctx, "context deadline exceeded", log.Error(ctx.Err()))
+	} else {
+		log.Event(ctx, "graceful shutdown complete", log.Data{"context": ctx.Err()})
 	}
-	log.Event(nil, "graceful shutdown of http server complete", nil)
-	hc.Stop()
-	log.Event(nil, "shutdown complete", nil)
-	cancel()
+
```
Comment (daiLlew) [inline main.go:105]: 👇 Should the os exit code always be 1?

**Comment by daiLlew** [comment]
Just a question on the exit code otherwise LGTM

File: main.go (hunk 1 above)
Comment (nshumoogum) [reply main.go:105]: You are right, it should be 0 as it successfully shutdown - I will update

**Review by daiLlew — APPROVED** [review]

**Comment by DavidSubiros** [comment]
I tried to run the tests and validated the changes using postman. Merging.

**Pull request merged**